// pokenode-ts/dist/constants/items.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemPockets = exports.ItemFlingEffects = exports.ItemCategories = exports.ItemAttributes = void 0;
var ItemAttributes;
(function (ItemAttributes) {
    ItemAttributes[ItemAttributes["COUNTABLE"] = 1] = "COUNTABLE";
    ItemAttributes[ItemAttributes["CONSUMABLE"] = 2] = "CONSUMABLE";
    ItemAttributes[ItemAttributes["USABLE_OVERWORLD"] = 3] = "USABLE_OVERWORLD";
    ItemAttributes[ItemAttributes["USABLE_IN_BATTLE"] = 4] = "USABLE_IN_BATTLE";
    ItemAttributes[ItemAttributes["HOLDABLE"] = 5] = "HOLDABLE";
    ItemAttributes[ItemAttributes["HOLDABLE_PASSIVE"] = 6] = "HOLDABLE_PASSIVE";
    ItemAttributes[ItemAttributes["HOLDABLE_ACTIVE"] = 7] = "HOLDABLE_ACTIVE";
    ItemAttributes[ItemAttributes["UNDERGROUND"] = 8] = "UNDERGROUND";
})(ItemAttributes = exports.ItemAttributes || (exports.ItemAttributes = {}));
var ItemCategories;
(function (ItemCategories) {
    ItemCategories[ItemCategories["STAT_BOOSTS"] = 1] = "STAT_BOOSTS";
    ItemCategories[ItemCategories["EFFORT_DROP"] = 2] = "EFFORT_DROP";
    ItemCategories[ItemCategories["MEDICINE"] = 3] = "MEDICINE";
    ItemCategories[ItemCategories["OTHER"] = 4] = "OTHER";
    ItemCategories[ItemCategories["IN_A_PINCH"] = 5] = "IN_A_PINCH";
    ItemCategories[ItemCategories["PICKY_HEALING"] = 6] = "PICKY_HEALING";
    ItemCategories[ItemCategories["TYPE_PROTECTION"] = 7] = "TYPE_PROTECTION";
    ItemCategories[ItemCategories["BAKING_ONLY"] = 8] = "BAKING_ONLY";
    ItemCategories[ItemCategories["COLLECTIBLES"] = 9] = "COLLECTIBLES";
    ItemCategories[ItemCategories["EVOLUTION"] = 10] = "EVOLUTION";
    ItemCategories[ItemCategories["SPELUNKING"] = 11] = "SPELUNKING";
    ItemCategories[ItemCategories["HELD_ITEMS"] = 12] = "HELD_ITEMS";
    ItemCategories[ItemCategories["CHOICE"] = 13] = "CHOICE";
    ItemCategories[ItemCategories["EFFORT_TRAINING"] = 14] = "EFFORT_TRAINING";
    ItemCategories[ItemCategories["BAD_HELD_ITEMS"] = 15] = "BAD_HELD_ITEMS";
    ItemCategories[ItemCategories["TRAINING"] = 16] = "TRAINING";
    ItemCategories[ItemCategories["PLATES"] = 17] = "PLATES";
    ItemCategories[ItemCategories["SPECIES_SPECIFIC"] = 18] = "SPECIES_SPECIFIC";
    ItemCategories[ItemCategories["TYPE_ENHANCEMENT"] = 19] = "TYPE_ENHANCEMENT";
    ItemCategories[ItemCategories["EVENT_ITEMS"] = 20] = "EVENT_ITEMS";
    ItemCategories[ItemCategories["GAMEPLAY"] = 21] = "GAMEPLAY";
    ItemCategories[ItemCategories["PLOT_ADVANCEMENT"] = 22] = "PLOT_ADVANCEMENT";
    ItemCategories[ItemCategories["UNUSED"] = 23] = "UNUSED";
    ItemCategories[ItemCategories["LOOT"] = 24] = "LOOT";
    ItemCategories[ItemCategories["ALL_MAIL"] = 25] = "ALL_MAIL";
    ItemCategories[ItemCategories["VITAMINS"] = 26] = "VITAMINS";
    ItemCategories[ItemCategories["HEALING"] = 27] = "HEALING";
    ItemCategories[ItemCategories["PP_RECOVERY"] = 28] = "PP_RECOVERY";
    ItemCategories[ItemCategories["REVIVAL"] = 29] = "REVIVAL";
    ItemCategories[ItemCategories["STATUS_CURES"] = 30] = "STATUS_CURES";
    ItemCategories[ItemCategories["MULCH"] = 32] = "MULCH";
    ItemCategories[ItemCategories["SPECIAL_BALLS"] = 33] = "SPECIAL_BALLS";
    ItemCategories[ItemCategories["STANDARD_BALLS"] = 34] = "STANDARD_BALLS";
    ItemCategories[ItemCategories["DEX_COMPLETION"] = 35] = "DEX_COMPLETION";
    ItemCategories[ItemCategories["SCARVES"] = 36] = "SCARVES";
    ItemCategories[ItemCategories["ALL_MACHINES"] = 37] = "ALL_MACHINES";
    ItemCategories[ItemCategories["FLUTES"] = 38] = "FLUTES";
    ItemCategories[ItemCategories["APRICORN_BALLS"] = 39] = "APRICORN_BALLS";
    ItemCategories[ItemCategories["APRICORN_BOX"] = 40] = "APRICORN_BOX";
    ItemCategories[ItemCategories["DATA_CARDS"] = 41] = "DATA_CARDS";
    ItemCategories[ItemCategories["JEWELS"] = 42] = "JEWELS";
    ItemCategories[ItemCategories["MIRACLE_SHOOTER"] = 43] = "MIRACLE_SHOOTER";
    ItemCategories[ItemCategories["MEGA_STONES"] = 44] = "MEGA_STONES";
    ItemCategories[ItemCategories["MEMORIES"] = 45] = "MEMORIES";
    ItemCategories[ItemCategories["Z_CRYSTALS"] = 46] = "Z_CRYSTALS";
})(ItemCategories = exports.ItemCategories || (exports.ItemCategories = {}));
var ItemFlingEffects;
(function (ItemFlingEffects) {
    ItemFlingEffects[ItemFlingEffects["BADLY_POISON"] = 1] = "BADLY_POISON";
    ItemFlingEffects[ItemFlingEffects["BURN"] = 2] = "BURN";
    ItemFlingEffects[ItemFlingEffects["BERRY_EFFECT"] = 3] = "BERRY_EFFECT";
    ItemFlingEffects[ItemFlingEffects["HERB_EFFECT"] = 4] = "HERB_EFFECT";
    ItemFlingEffects[ItemFlingEffects["PARALYZE"] = 5] = "PARALYZE";
    ItemFlingEffects[ItemFlingEffects["POISON"] = 6] = "POISON";
    ItemFlingEffects[ItemFlingEffects["FLINCH"] = 7] = "FLINCH";
})(ItemFlingEffects = exports.ItemFlingEffects || (exports.ItemFlingEffects = {}));
var ItemPockets;
(function (ItemPockets) {
    ItemPockets[ItemPockets["MISC"] = 1] = "MISC";
    ItemPockets[ItemPockets["MEDICINE"] = 2] = "MEDICINE";
    ItemPockets[ItemPockets["POKEBALLS"] = 3] = "POKEBALLS";
    ItemPockets[ItemPockets["MACHINES"] = 4] = "MACHINES";
    ItemPockets[ItemPockets["BERRIES"] = 5] = "BERRIES";
    ItemPockets[ItemPockets["MAIL"] = 6] = "MAIL";
    ItemPockets[ItemPockets["BATTLE"] = 7] = "BATTLE";
    ItemPockets[ItemPockets["KEY"] = 8] = "KEY";
})(ItemPockets = exports.ItemPockets || (exports.ItemPockets = {}));
