// pokenode-ts/dist/constants/moves.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MoveTargets = exports.MoveLearnMethods = exports.MoveDamageClasses = exports.MoveCategories = exports.MoveBattleStyles = exports.MoveAilmtents = void 0;
var MoveAilmtents;
(function (MoveAilmtents) {
    MoveAilmtents[MoveAilmtents["UNKNOWN"] = -1] = "UNKNOWN";
    MoveAilmtents[MoveAilmtents["NONE"] = 0] = "NONE";
    MoveAilmtents[MoveAilmtents["PARALYSIS"] = 1] = "PARALYSIS";
    MoveAilmtents[MoveAilmtents["SLEEP"] = 2] = "SLEEP";
    MoveAilmtents[MoveAilmtents["FREEZE"] = 3] = "FREEZE";
    MoveAilmtents[MoveAilmtents["BURN"] = 4] = "BURN";
    MoveAilmtents[MoveAilmtents["POISON"] = 5] = "POISON";
    MoveAilmtents[MoveAilmtents["CONFUSION"] = 6] = "CONFUSION";
    MoveAilmtents[MoveAilmtents["INFATUATION"] = 7] = "INFATUATION";
    MoveAilmtents[MoveAilmtents["TRAP"] = 8] = "TRAP";
    MoveAilmtents[MoveAilmtents["NIGHTMARE"] = 9] = "NIGHTMARE";
    MoveAilmtents[MoveAilmtents["TORMENT"] = 12] = "TORMENT";
    MoveAilmtents[MoveAilmtents["DISABLE"] = 13] = "DISABLE";
    MoveAilmtents[MoveAilmtents["YAWN"] = 14] = "YAWN";
    MoveAilmtents[MoveAilmtents["HEAL_BLOCK"] = 15] = "HEAL_BLOCK";
    MoveAilmtents[MoveAilmtents["NO_TYPE_IMMUNITY"] = 17] = "NO_TYPE_IMMUNITY";
    MoveAilmtents[MoveAilmtents["LEECH_SEED"] = 18] = "LEECH_SEED";
    MoveAilmtents[MoveAilmtents["EMBARGO"] = 19] = "EMBARGO";
    MoveAilmtents[MoveAilmtents["PERISH_SONG"] = 20] = "PERISH_SONG";
    MoveAilmtents[MoveAilmtents["INGRAIN"] = 21] = "INGRAIN";
    MoveAilmtents[MoveAilmtents["SILENCE"] = 24] = "SILENCE";
})(MoveAilmtents = exports.MoveAilmtents || (exports.MoveAilmtents = {}));
var MoveBattleStyles;
(function (MoveBattleStyles) {
    MoveBattleStyles[MoveBattleStyles["ATTACK"] = 1] = "ATTACK";
    MoveBattleStyles[MoveBattleStyles["DEFENSE"] = 2] = "DEFENSE";
    MoveBattleStyles[MoveBattleStyles["SUPPORT"] = 3] = "SUPPORT";
})(MoveBattleStyles = exports.MoveBattleStyles || (exports.MoveBattleStyles = {}));
var MoveCategories;
(function (MoveCategories) {
    MoveCategories[MoveCategories["DAMAGE"] = 0] = "DAMAGE";
    MoveCategories[MoveCategories["AILMENT"] = 1] = "AILMENT";
    MoveCategories[MoveCategories["NET_GOOD_STATS"] = 2] = "NET_GOOD_STATS";
    MoveCategories[MoveCategories["HEAL"] = 3] = "HEAL";
    MoveCategories[MoveCategories["DAMAGE_AILMENT"] = 4] = "DAMAGE_AILMENT";
    MoveCategories[MoveCategories["SWAGGER"] = 5] = "SWAGGER";
    MoveCategories[MoveCategories["DAMAGE_LOWER"] = 6] = "DAMAGE_LOWER";
    MoveCategories[MoveCategories["DAMAGE_RAISE"] = 7] = "DAMAGE_RAISE";
    MoveCategories[MoveCategories["DAMAGE_HEAL"] = 8] = "DAMAGE_HEAL";
    MoveCategories[MoveCategories["OHKO"] = 9] = "OHKO";
    MoveCategories[MoveCategories["WHOLE_FIELD_EFFECT"] = 10] = "WHOLE_FIELD_EFFECT";
    MoveCategories[MoveCategories["FIELD_EFFECT"] = 11] = "FIELD_EFFECT";
    MoveCategories[MoveCategories["FORCE_SWITCH"] = 12] = "FORCE_SWITCH";
    MoveCategories[MoveCategories["UNIQUE"] = 13] = "UNIQUE";
})(MoveCategories = exports.MoveCategories || (exports.MoveCategories = {}));
var MoveDamageClasses;
(function (MoveDamageClasses) {
    MoveDamageClasses[MoveDamageClasses["STATUS"] = 1] = "STATUS";
    MoveDamageClasses[MoveDamageClasses["PHYSICAL"] = 2] = "PHYSICAL";
    MoveDamageClasses[MoveDamageClasses["SPECIAL"] = 3] = "SPECIAL";
})(MoveDamageClasses = exports.MoveDamageClasses || (exports.MoveDamageClasses = {}));
var MoveLearnMethods;
(function (MoveLearnMethods) {
    MoveLearnMethods[MoveLearnMethods["LEVEL_UP"] = 1] = "LEVEL_UP";
    MoveLearnMethods[MoveLearnMethods["EGG"] = 2] = "EGG";
    MoveLearnMethods[MoveLearnMethods["TUTOR"] = 3] = "TUTOR";
    MoveLearnMethods[MoveLearnMethods["MACHINE"] = 4] = "MACHINE";
    MoveLearnMethods[MoveLearnMethods["STADIUM_SURFING_PIKACHU"] = 5] = "STADIUM_SURFING_PIKACHU";
    MoveLearnMethods[MoveLearnMethods["LIGHT_BALL_EGG"] = 6] = "LIGHT_BALL_EGG";
    MoveLearnMethods[MoveLearnMethods["COLOSSEUM_PURIFICATION"] = 7] = "COLOSSEUM_PURIFICATION";
    MoveLearnMethods[MoveLearnMethods["XD_SHADOW"] = 8] = "XD_SHADOW";
    MoveLearnMethods[MoveLearnMethods["XD_PURIFICATION"] = 9] = "XD_PURIFICATION";
    MoveLearnMethods[MoveLearnMethods["FORM_CHANGE"] = 10] = "FORM_CHANGE";
})(MoveLearnMethods = exports.MoveLearnMethods || (exports.MoveLearnMethods = {}));
var MoveTargets;
(function (MoveTargets) {
    MoveTargets[MoveTargets["SPECIFIC_MOVE"] = 1] = "SPECIFIC_MOVE";
    MoveTargets[MoveTargets["SELECTED_POKEMON_ME_FIRST"] = 2] = "SELECTED_POKEMON_ME_FIRST";
    MoveTargets[MoveTargets["ALLY"] = 3] = "ALLY";
    MoveTargets[MoveTargets["USERS_FIELD"] = 4] = "USERS_FIELD";
    MoveTargets[MoveTargets["USER_OR_ALLY"] = 5] = "USER_OR_ALLY";
    MoveTargets[MoveTargets["OPPONENTS_FIELD"] = 6] = "OPPONENTS_FIELD";
    MoveTargets[MoveTargets["USER"] = 7] = "USER";
    MoveTargets[MoveTargets["RANDOM_OPPONENT"] = 8] = "RANDOM_OPPONENT";
    MoveTargets[MoveTargets["ALL_OTHER_POKEMON"] = 9] = "ALL_OTHER_POKEMON";
    MoveTargets[MoveTargets["SELECTED_POKEMON"] = 10] = "SELECTED_POKEMON";
    MoveTargets[MoveTargets["ALL_OPPONENTS"] = 11] = "ALL_OPPONENTS";
    MoveTargets[MoveTargets["ENTIRE_FIELD"] = 12] = "ENTIRE_FIELD";
    MoveTargets[MoveTargets["USER_AND_ALIES"] = 13] = "USER_AND_ALIES";
    MoveTargets[MoveTargets["ALL_POKEMON"] = 14] = "ALL_POKEMON";
    MoveTargets[MoveTargets["ALL_ALLIES"] = 15] = "ALL_ALLIES";
})(MoveTargets = exports.MoveTargets || (exports.MoveTargets = {}));
